### PR TITLE
W-15884079: Improve encapsulation of the Embedded Runtime

### PR DESCRIPTION
**Pull request by rbourbonnavarro**

### CODE REVIEW

File: embedded/embedded-test/tests/src/main/java/module-info.java [hunk 1]
```diff
@@ -4,24 +4,17 @@
  * license, a copy of which has been included with this distribution in the
  * LICENSE.txt file.
  */
+
+
 /**
  * Mule Embedded Implementation Tests.
  *
- * @moduleGraph
```
Comment (elrodro83) [inline embedded/embedded-test/tests/src/main/java/module-info.java:10]: ?

File: embedded/embedded-test/tests/src/test/java/org/mule/runtime/module/embedded/AbstractEmbeddedTestCase.java [hunk 2]
```diff
@@ -72,6 +76,14 @@ public static void dispose() {
   public SystemProperty simpleLogging =
       new SystemProperty("mule.simpleLog", "true");
 
+  @Parameter
+  public boolean useIsolation;
+
+  @Parameters(name = "{0}")
```
Comment (elrodro83) [inline embedded/embedded-test/tests/src/test/java/org/mule/runtime/module/embedded/AbstractEmbeddedTestCase.java:82]: `useIsolation: {0}`

File: embedded/embedded-test/tests/src/test/java/org/mule/runtime/module/embedded/EmbeddedLifecycleTestCase.java [hunk 3]
```diff
@@ -64,8 +71,22 @@ public static void initialise() {
...
+  // mule-embedded-api no longer supports log4j configuration since 1.7, this property has to be set so the log4j configuration
+  // doesn't take place
+  @Rule
+  public SystemProperty simpleLogging =
+      new SystemProperty("mule.simpleLog", "true");
+
+  @Parameter
+  public boolean useIsolation;
+
+  @Parameters(name = "{0}")
```
Comment (elrodro83) [inline embedded/embedded-test/tests/src/test/java/org/mule/runtime/module/embedded/EmbeddedLifecycleTestCase.java:83]: `useIsolation: {0}`

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: embedded/embedded-test/tests/src/main/java/module-info.java (hunk 1 above)
Comment (rbourbonnavarro) [reply embedded/embedded-test/tests/src/main/java/module-info.java:10]: Restored.

File: embedded/embedded-test/tests/src/test/java/org/mule/runtime/module/embedded/EmbeddedLifecycleTestCase.java (hunk 3 above)
Comment (rbourbonnavarro) [reply embedded/embedded-test/tests/src/test/java/org/mule/runtime/module/embedded/EmbeddedLifecycleTestCase.java:83]: Done.

File: embedded/embedded-test/tests/src/test/java/org/mule/runtime/module/embedded/AbstractEmbeddedTestCase.java (hunk 2 above)
Comment (rbourbonnavarro) [reply embedded/embedded-test/tests/src/test/java/org/mule/runtime/module/embedded/AbstractEmbeddedTestCase.java:82]: Done.

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**